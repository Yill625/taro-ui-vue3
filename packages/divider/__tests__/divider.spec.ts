import {
  hexToRGBA,
  genMountFn,
  testPropClassAndStyle
} from '@taro-ui-vue3/test-utils/helper'
import {
  pxTransformMockFn
} from '@taro-ui-vue3/test-utils/@tarojs/taro'
import AtDivider from '../index.vue'

const mountFn = genMountFn(AtDivider)

describe('AtDivider', () => {
  testPropClassAndStyle(mountFn)

  it('should render default props and match snapshot', () => {
    const wrapper = mountFn()
    expect(wrapper.element).toMatchSnapshot()
  })

  it.concurrent('should render prop content', async () => {
    const wrapper = mountFn({ content: 'content' })
    expect(
      wrapper
        .find('.at-divider__content > view')
        .text()
    ).toEqual('content')
  })

  it.concurrent('should render slot content', async () => {
    const wrapper = mountFn({}, {
      default: ['content']
    })
    expect(
      wrapper
        .find('.at-divider__content')
        .text()
    ).toEqual('content')
  })

  it.concurrent('should render prop height', async () => {
    const wrapper = mountFn({ height: '120' })
    expect(
      wrapper.attributes('style')
    ).toEqual(`height: ${pxTransformMockFn(120)};`)
  })

  it.concurrent.each([
    ['fontColor', '#fff'],
    ['fontSize', 56]
  ])('should render prop %s', async (prop, value) => {
    const wrapper = mountFn({ [prop]: value })

    const expected = prop === 'fontColor'
      ? `color: ${hexToRGBA(value as string)};`
      : `font-size: ${pxTransformMockFn(value)};`

    expect(
      wrapper
        .find('.at-divider__content')
        .attributes('style')
    ).toContain(expected)
  })

  it.concurrent('should render prop lineColor', async () => {
    const wrapper = mountFn({ lineColor: '#ddd' })

    expect(
      wrapper
        .find('.at-divider__line')
        .attributes('style')
    ).toContain(`background-color: ${hexToRGBA('#ddd')};`)

  })
})
