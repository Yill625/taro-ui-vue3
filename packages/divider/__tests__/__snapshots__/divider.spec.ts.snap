// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtDivider should render default AtDivider 1`] = `
<view
  class="at-divider"
>
  <view
    class="at-divider__content"
  />
  <view
    class="at-divider__line"
  />
</view>
`;

exports[`AtDivider should render prop -- childen 1`] = `
<view
  childen="content"
  class="at-divider"
>
  <view
    class="at-divider__content"
  >
    test
  </view>
  <view
    class="at-divider__line"
  />
</view>
`;

exports[`AtDivider should render prop -- class 1`] = `
<view
  class="test at-divider test"
>
  <view
    class="at-divider__content"
  />
  <view
    class="at-divider__line"
  />
</view>
`;

exports[`AtDivider should render prop -- content 1`] = `
<view
  class="at-divider"
>
  <view
    class="at-divider__content"
  >
    content
  </view>
  <view
    class="at-divider__line"
  />
</view>
`;

exports[`AtDivider should render prop -- fontColor 1`] = `
<view
  class="at-divider"
>
  <view
    class="at-divider__content"
    style="color: rgb(255, 255, 255);"
  />
  <view
    class="at-divider__line"
  />
</view>
`;

exports[`AtDivider should render prop -- fontSize 1`] = `
<view
  class="at-divider"
>
  <view
    class="at-divider__content"
    style="font-size: 1.1947rem;"
  />
  <view
    class="at-divider__line"
  />
</view>
`;

exports[`AtDivider should render prop -- height 1`] = `
<view
  class="at-divider"
  style="height: 2.56rem;"
>
  <view
    class="at-divider__content"
  />
  <view
    class="at-divider__line"
  />
</view>
`;

exports[`AtDivider should render prop -- lineColor 1`] = `
<view
  class="at-divider"
>
  <view
    class="at-divider__content"
  />
  <view
    class="at-divider__line"
    style="background-color: rgb(255, 255, 255);"
  />
</view>
`;

exports[`AtDivider should render prop -- style 1`] = `
<view
  class="at-divider"
  style="color: red;"
>
  <view
    class="at-divider__content"
  />
  <view
    class="at-divider__line"
  />
</view>
`;
